testsuite/features/support/env.rb

```diff
@@ -73,6 +74,10 @@
 $beta_enabled = (ENV.fetch('BETA_ENABLED', 'False') == 'True')
 $api_protocol = ENV.fetch('API_PROTOCOL', nil) if ENV['API_PROTOCOL'] # force the API protocol to be used. You can use 'http' or 'xmlrpc'
 
+# Define a global counter to track consecutive ReadTimeout errors
+$timeout_failure_count = 0
+$timeout_threshold = 10  # Set the threshold for stopping the suite
```

Comment:
This value does not aim to change during the whole cucumber execution, so I would better define it as a constant than a global variable. Also, if we have it as a global variable, please move it up to the block of lines 45-59